hub/core/dataset/dataset.py

```diff
@@ -239,26 +239,29 @@ def __getitem__(
         is_iteration: bool = False,
     ):
         if isinstance(item, str):
-            fullpath = posixpath.join(self.group_index, item)
-            tensor = self._get_tensor_from_root(fullpath)
-            if tensor is not None:
-                return tensor[self.index]
-            elif self._has_group_in_root(fullpath):
-                return self.__class__(
-                    storage=self.storage,
-                    index=self.index,
-                    group_index=posixpath.join(self.group_index, item),
-                    read_only=self.read_only,
-                    token=self._token,
-                    verbose=False,
-                    version_state=self.version_state,
-                    path=self.path,
-                )
-            elif "/" in item:
-                splt = posixpath.split(item)
-                return self[splt[0]][splt[1]]
+            if(self.index > self.__len__):
```

Comment:
DO NOT push code that you have never run yourself locally. Index object doesn't support `>` comparison, and `self.__len__` is a method, not a number. This line of code will practically fail *all* unit tests.

Reply:
I should store the index and len in a variable and then compare it.

Reply:
Like `len=__len__(item)`